src/vi/nhattruyen/src/eu/kanade/tachiyomi/extension/vi/nhattruyen/NhatTruyen.kt

```diff
@@ -177,4 +187,32 @@ class NhatTruyen : WPComics(
             OrderFilter(),
         )
     }
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val url = client.newCall(GET(baseUrl + manga.url, headers)).execute().asJsoup()
+        if (checkChapterLists(url).isNotEmpty()) {
+            val slug = manga.url.removePrefix("/truyen-tranh/")
+            return client.newCall(GET("$baseUrl/Comic/Services/ComicService.asmx/ChapterList?slug=$slug", headers))
+                .asObservableSuccess()
+                .map { response -> chapterListParse(response) }
+        }
+        return super.fetchChapterList(manga)
+    }
+
+    private fun checkChapterLists(document: Document) = document.selectFirst("a.view-more.hidden")!!.text()
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val json = json.decodeFromString<ChapterDTO>(response.body.string())
```

Comment:
```suggestion
        val json = response.parseAs<ChapterDTO>()
```
need to import, and remove the json by injectLazy above too